pkg/controller/vpc.go

```diff
@@ -1070,6 +1070,10 @@ func (c *Controller) getVpcSubnets(vpc *kubeovnv1.Vpc) (subnets []string, defaul
 		if subnet.Spec.Default {
 			defaultSubnet = subnet.Name
 		}
+
+		if vpc.Name != util.DefaultVpc && vpc.Spec.DefaultSubnet != "" && vpc.Spec.DefaultSubnet == subnet.Name {
```

Comment:
is it better to make sure all vpc has the spec `DefaultSubnet`?

Reply:
wdym? every vpc should have defaultSubnet and be validated?
I was thinking of having it as an optional field in the first iteration

Reply:
yes, every vpc should have its defaultSubnet, in the ovn-cluster, the  defaultSubnet is ovn default.

Reply:
I think, it is better to keep the `defaultSubnet logic` just the same in both default and custom vpc.

Reply:
For custom VPC there won't be a default subnet, until you specify vpc.spec.defaultSubnet
We could set it for ovn-cluster, but I think there is another logic that handles default vpc in the namespace.go

Reply:
ok，`vpc has defaultSubnet` looks good to me